power_grid_model_c/power_grid_model/include/power_grid_model/component/asym_line.hpp

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include <numeric>
+
+#include <iostream>
+#include "branch.hpp"
+
+#include "../auxiliary/input.hpp"
+#include "../auxiliary/output.hpp"
+#include "../auxiliary/update.hpp"
+#include "../calculation_parameters.hpp"
+#include "../common/common.hpp"
+#include "../common/three_phase_tensor.hpp"
+
+namespace power_grid_model {
+
+class AsymLine : public Branch {
+  public:
+    using InputType = AsymLineInput;
+    using UpdateType = BranchUpdate;
+    static constexpr char const* name = "asym_line";
+
+    explicit AsymLine(AsymLineInput const& asym_line_input, double system_frequency, double u1, double u2)
+        : Branch{asym_line_input}, i_n_{asym_line_input.i_n}, base_i_{base_power_3p / u1 / sqrt3} {
+        if (cabs(u1 - u2) > numerical_tolerance) {
+            throw ConflictVoltage{id(), from_node(), to_node(), u1, u2};
+        }
+
+        ComplexTensor<asymmetric_t> c_matrix = compute_c_matrix_from_input(asym_line_input);
+        ComplexTensor<asymmetric_t> z_series = compute_z_series_from_input(asym_line_input);
+
+        ComplexTensor<asymmetric_t> empty_tensor;
+
+        if (c_matrix.isZero() || z_series.isZero()) {
+            throw UnsupportedInputDescriptionAsymLine();
+        }
```

Comment:
as mentioned elsewhere: we generally do not do input data value checks in the core. it just triggers UB (undefined/unspecified behavior). E.g. in `Line`, we also do not check that `z` is non-zero. Instead, we just accept `NaN` values, which, in turn, propagate to the user output. The user can then see in the data that it is corrupted.